components/button/style.scss

```diff
@@ -1,10 +1,143 @@
+		background: #f7f7f7;
+		box-shadow: inset 0 -1px 0 #cccccc;
+		vertical-align: top;
+
+		&:hover {
+			background: #fafafa;
+			border-color: #999;
+			box-shadow: inset 0 -1px 0 #999;
+			color: #23282d;
+		}
+
+		&:focus:not(:disabled):not([aria-disabled="true"]) {
+			background: #fafafa;
+			color: #23282d;
+			border-color: #999;
+			box-shadow: inset 0 -1px 0 #999,
+				0 0 0 2px $blue-medium-200;
+		}
+
+		&:active:not(:disabled):not([aria-disabled="true"]) {
+			background: #eee;
+			border-color: #999;
+			box-shadow: inset 0 1px 0 #999,
+		}
+
+		&:disabled,
+		&[disabled] {
+			color: #a0a5aa !important;
+			border-color: #ddd !important;
+			background: #f7f7f7 !important;
+			box-shadow: none !important;
+			text-shadow: 0 1px 0 #fff !important;
+			cursor: default;
+			-webkit-transform: none !important;
+			transform: none !important;
+		}
+	}
+
+	&.is-primary {
+		background: color( theme( button ) shade( 10% ) );
+		border-color: color( theme( button ) shade( 20% ) ) color( theme( button ) shade( 30% ) ) color( theme( primary ) shade( 30% ) );
+		box-shadow: inset 0 -1px 0 color( theme( button ) shade( 30% ) );
+		color: #fff;
+		text-decoration: none;
+		text-shadow: 0 -1px 1px color( theme( button ) shade( 30% ) ),
+			1px 0 1px color( theme( button ) shade( 30% ) ),
+			0 1px 1px color( theme( button ) shade( 30% ) ),
+			-1px 0 1px color( theme( button ) shade( 30% ) );
+
+		&:hover,
+		&:focus:not(:disabled):not([aria-disabled="true"]) {
+			background: color( theme( button ) shade( 15% ) );
+			border-color: color( theme( button ) shade( 50% ) );
+			box-shadow: inset 0 -1px 0 color( theme( button ) shade( 50% ) );
+			color: $white;
+		}
+
+		&:focus:not(:disabled):not([aria-disabled="true"]) {
+			box-shadow: inset 0 -1px 0 color( theme( button ) shade( 50% ) ),
+				0 0 0 2px $blue-medium-200;
+		}
+
+		&:active:not(:disabled):not([aria-disabled="true"]) {
+			background: color( theme( button ) shade( 20% ) );
+			border-color: color( theme( button ) shade( 50% ) );
+			box-shadow: inset 0 1px 0 color( theme( button ) shade( 50% ) );
+			vertical-align: top;
+		}
+
+		&:disabled,
+		&[disabled] {
+			color: color( theme( button ) tint( 30% ) ) !important;
+			background: color( theme( button ) shade( 30% ) ) !important;
+			border-color: color( theme( button ) shade( 20% ) ) !important;
+			box-shadow: none !important;
+			text-shadow: 0 -1px 0 rgba( 0, 0, 0, 0.1 ) !important;
+			cursor: default;
+		}
+
+		&.is-busy,
+		&.is-primary.is-busy[disabled] {
+			color: $white !important;
+			background-size: 100px 100% !important;
+			background-image: repeating-linear-gradient( -45deg, theme( primary ), color( theme( primary ) shade( 50% ) ) 11px, color( theme( primary ) shade( 50% ) ) 10px, theme( primary ) 20px) !important;
+			border-color: color( theme( primary ) shade( 50% ) ) !important;
+		}
+	}
+
+	/* Buttons that look like links, for a cross of good semantics with the visual */
+	&.is-link {
+		margin: 0;
```

Comment:
Regression: This selector is more specific than (well, equal specific but out of order from) and thus overrides the margin applied from `.components-color-palette_clear`:

https://github.com/WordPress/gutenberg/blob/94889af43baf5142c52f79190947492bc86f9567/components/color-palette/style.scss#L9

v4.9.2|#6562
---|---
![v4.9.2](https://user-images.githubusercontent.com/1779930/40931233-55e74722-67f8-11e8-99f8-86613cb74364.png)|![#6562](https://user-images.githubusercontent.com/1779930/40931242-619d5bb0-67f8-11e8-8a39-0bba2b9c8c03.png)

Reply:
Honestly, we shouldn't need margin here for the button. The color palette panel shouldn't be doing a negative margin. It should have a container and spread the options via `display: flex` & `align-items: space-between`